File: app/src/main/java/it/cammino/risuscito/database/dao/LocalLinksDao.java
package it.cammino.risuscito.database.dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;

import it.cammino.risuscito.database.entities.LocalLink;

import static android.arch.persistence.room.OnConflictStrategy.REPLACE;

@Dao
public interface LocalLinksDao {

  @Query("SELECT * FROM locallink WHERE idCanto = :id")
  LocalLink getLocalLinkByCantoId(int id);

  @Update
  void updateLocalLink(LocalLink localLink);

  @Insert(onConflict = REPLACE)
  void updateLocalLink(List<LocalLink> localLinkList);

  @Delete
  void deleteLocalLink(LocalLink localLink);

  @Delete
  void deleteLocalLink(List<LocalLink> localLinkList);

  @Insert
  void insertLocalLink(LocalLink localLink);

  @Insert
  void insertLocalLink(List<LocalLink> localLinkList);
}

File: app/src/main/java/it/cammino/risuscito/database/dao/LocalLinksDao.kt
package it.cammino.risuscito.database.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import androidx.room.Upsert
import it.cammino.risuscito.database.entities.LocalLink

@Dao
interface LocalLinksDao {

    @Query("DELETE FROM locallink")
    fun truncateTable()

    @Query("SELECT * FROM locallink")
    fun all(): List<LocalLink>

    @Query("SELECT * FROM locallink WHERE idCanto = :id")
    fun getLocalLinkByCantoId(id: Int): LocalLink?

    @Update
    fun updateLocalLink(localLink: LocalLink)

    @Upsert
    fun updateLocalLink(localLinkList: List<LocalLink>)

    @Delete
    fun deleteLocalLink(localLink: LocalLink)

    @Delete
    fun deleteLocalLink(localLinkList: List<LocalLink>)

    @Insert
    fun insertLocalLink(localLink: LocalLink)

    @Insert
    fun insertLocalLink(localLinkList: List<LocalLink>)
}
